sql/core/src/main/scala/org/apache/spark/sql/execution/UnsafeRowSerializer.scala

```diff
@@ -143,7 +143,7 @@ private class UnsafeRowSerializerInstance(numFields: Int) extends SerializerInst
       override def readKey[T: ClassTag](): T = {
         // We skipped serialization of the key in writeKey(), so just return a dummy value since
         // this is going to be discarded anyways.
-        null.asInstanceOf[T]
+        (-1).asInstanceOf[T]
```

Comment:
then of course we need to relax the assertion in `writeKey` and add a comment that explains why we need to accept `key == null` there.